test/treasure.utils.spec.js

```diff
@@ -23,4 +25,37 @@ describe('Treasure Utils', function () {
       expect(uuidRegex.test(generateUUID())).to.equal(true)
     })
   })
+
+  describe('misc', function () {
+    describe('fetchWithTimeout', function () {
+      it('must abort after a while', function (done) {
+        if (window.AbortController) {
+          var abortSpy = simple.mock(window.AbortController.prototype, 'abort')
+          fetchWithTimeout('https://apple.com', 1)
+            .then(function () {
+              done(new Error('but instead succeeded'))
+            })['catch'](function () {
+              if (abortSpy.callCount === 1) {
```

Comment:
Can we use `expect` here?

Reply:
We can't because it will throw Exception on failure case, which will be swallowed and will not reach Promise.finally causing false positive test.